airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py

```diff
@@ -166,7 +165,7 @@ def create_job_name(self):
 
         updated_name = add_unique_suffix(name=name, max_len=MAX_LABEL_LEN)
 
-        return re.sub(r"[^a-z0-9-]+", "-", updated_name.lower())
+        return self._set_name(updated_name.lower())
```

Comment:
I think you don't have to do `lower()` it will do it in the function

Reply:
ah sorry my bad, i have updated now.